test/server/controllers/ArticleController.test.js

```diff
@@ -0,0 +1,136 @@
+import chaiHttp from 'chai-http';
+import chai from 'chai';
+import app from '../../../app';
+import {
+  articleInputValid,
+  articleInputNoTitle,
+} from '../../../mockdata/articleMockData';
+
+chai.should();
+
+chai.use(chaiHttp);
+
+describe('Articles Controller Tests', () => {
+  const userData = {};
+  before((done) => {
+    // Sign up a user and get the id and token returned
+    chai.request(app)
+      .post('/api/v1/users/signup')
+      .send({
+        fullName: 'Not Tani',
+        email: 'n.tani@whowa.com',
+        password: 'ntanirfsee4',
+      })
+      .end((err, res) => {
+        userData.id = res.body.user.id;
+        userData.token = res.body.user.token;
```

Comment:
This looks good but we would want to reuse the id and token of this signed up user in other test files, may be to rate an article or give a comment. This way, we reduce the lines of code we write for a user-sign up test. Can you make your `userData` object accessible to all test files since it is housing the id and token of this user or better still, you could use a `process.env variable`?

Reply:
@augustineezinwa What i'm doing here is signing up a user, then using the token and id from the response to create an article.
I'm not sure I understand exactly what you mean by `reduce the lines of code we write for a user-sign up test`. Why do we need an environment variable to store the id and token of a user I created in a test file?

Reply:
I think the logic is okay like that. Whoever wants to use the token should sign in and get the token.